### ISSUE TITLE
Stripping out new line characters. 

### ISSUE THREAD

**Issue by garrows** [mathiasbynens/quoted-printable#2]
Hi,
It seems to be stripping out new line characters. Is this by design?

```
var in = 'top\nbottom';
var out = require('quoted-printable').decode(in);
console.log(out); //gives 'topbottom'
```

**Comment by mathiasbynens** [comment]
Yep. In quoted-printable encoding, new line characters should be encoded as `=0A` or as `=` followed by an actual new line character.

**Comment by garrows** [comment]
I don't think that's how a couple of mail clients I tested today did it.
I'll try to confirm tomorrow.

**Comment by mathiasbynens** [comment]
<https://tools.ietf.org/html/rfc2045#section-6.7>:

```
    (4)   (Line Breaks) A line break in a text body, represented
          as a CRLF sequence in the text canonical form, must be
          represented by a (RFC 822) line break, which is also a
          CRLF sequence, in the Quoted-Printable encoding.  Since
          the canonical representation of media types other than
          text do not generally include the representation of
          line breaks as CRLF sequences, no hard line breaks
          (i.e. line breaks that are intended to be meaningful
          and to be displayed to the user) can occur in the
          quoted-printable encoding of such types.  Sequences
          like "=0D", "=0A", "=0A=0D" and "=0D=0A" will routinely
          appear in non-text data represented in quoted-
          printable, of course.

          Note that many implementations may elect to encode the
          local representation of various content types directly
          rather than converting to canonical form first,
          encoding, and then converting back to local
          representation.  In particular, this may apply to plain
          text material on systems that use newline conventions
          other than a CRLF terminator sequence.  Such an
          implementation optimization is permissible, but only
          when the combined canonicalization-encoding step is
          equivalent to performing the three steps separately.
```

**Comment by garrows** [comment]
This is what iMail sends from the input of
```
hello⤵
world
```
gives 
```
"From: Glen Arrowsmith <glen.arrowsmith@gmail.com>
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable
Subject: Test
Message-Id: <44F0A215-9F1C-479A-B7EE-8E232D7487CB@itoc.com.au>
Date: Thu, 8 Jan 2015 19:05:20 +1000
To: Glen Arrowsmith <glen.arrowsmith@gmail.com>
Mime-Version: 1.0 (Mac OS X Mail 8.1 \(1993\))
X-Mailer: Apple Mail (2.1993)

hello=E2=A4=B5
world="
```

So the code
```
var i = 'hello=E2=A4=B5\r\nworld=';
var u = require('quoted-printable').decode(i);
var o = require('utf8').decode(u);
console.log(o); //prints hello⤵world=
```
Gives `hello⤵world=` which is obviously not what I'm after even though Apple is not following the standards (big suprise). 

I've already switched to using mimelib which handles it fine so feel free to leave this closed. 

Thanks

**Comment by mathiasbynens** [comment]
Thanks for the detailed info! I’ll take a look.

**Comment by mathiasbynens** [comment]
Actually, looks like I misinterpreted that part of the RFC. Fixed; thanks!

**Comment by mathiasbynens** [comment]
Just released v1.0.0 which includes this fix. Thanks again!
